File: common/src/main/java/com/jmsoftware/maf/common/domain/springbootstarter/HttpApiResourcesResponse.java
package com.jmsoftware.maf.common.domain.springbootstarter;

import com.google.common.collect.Lists;
import lombok.Data;
import org.springframework.web.bind.annotation.RequestMethod;

import java.util.List;

/**
 * Description: HttpApiResourcesResponse, change description here.
 *
 * @author 钟俊（zhongjun）, email: zhongjun@toguide.cn, date: 12/25/2020 4:58 PM
 **/
@Data
public class HttpApiResourcesResponse {
    private List<HttpApiResource> list = Lists.newLinkedList();

    @Data
    public static class HttpApiResource {
        private RequestMethod method;
        private String urlPattern;
    }
}

File: common/src/main/kotlin/com/jmsoftware/maf/common/domain/springbootstarter/HttpApiResourcesResponse.kt
package com.jmsoftware.maf.common.domain.springbootstarter

import org.springframework.web.bind.annotation.RequestMethod

/**
 * # HttpApiResourcesResponse
 *
 * Description: HttpApiResourcesResponse, change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/13/22 1:48 PM
 */
class HttpApiResourcesResponse {
    val list: MutableList<HttpApiResource> = mutableListOf()

    class HttpApiResource {
        var method: RequestMethod? = null
        var urlPattern: String? = null
    }
}
